src/App.tsx

```diff
@@ -1,12 +1,105 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { GoodsList } from './GoodsList/GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const [loadError, setLoadError] = useState(false);
+  const [choosed, setChoosed] = useState('');
 
-const App: React.FC = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+  const loadGoods = async () => {
+    try {
+      const loadedGoods = await getAll();
+
+      setGoods(loadedGoods);
+      setChoosed('You load All Goods');
+    } catch (error) {
+      setLoadError(true);
+    }
+  };
+
+  const loadFiveGoods = async () => {
+    try {
+      const loadedFiveGoods = await get5First();
+
+      setGoods(loadedFiveGoods);
+      setChoosed('You load First 5 Goods');
+    } catch (error) {
+      setLoadError(true);
+    }
+  };
+
+  const loadRedGoods = async () => {
+    try {
+      const loadedRedGoods = await getRedGoods();
+
+      setGoods(loadedRedGoods);
+      setChoosed('You load All Red Goods');
+    } catch (error) {
+      setLoadError(true);
+    }
+  };
```

Comment:
these 3 functions look pretty similar, let's pass the needed callback to ONE function and call it inside